### PR TITLE
Backend connection

### PR DESCRIPTION
**Pull request by IlanSirisky**
Added connection to backend with react-query.
Refactored all project to use the Pokemon interface from the backend.
Added debounce on search.

### CODE REVIEW

File: src/components/CardBody/CardBody.tsx [hunk 1]
```diff
@@ -34,7 +41,16 @@ const CardBody = ({
         </PowerContainer>
       </ImagePowerContainer>
       {subheadText && <StyledIdText>{subheadText}</StyledIdText>}
-      <HeadingLargeRegular>{cardTitle}</HeadingLargeRegular>
+      <StyledHeaderWrapper>
+        <HeadingLargeRegular>{cardTitle}</HeadingLargeRegular>
+        {iconFlag && bottomIcon && (
```
Comment (YuvalLeviMoveoHLS) [inline src/components/CardBody/CardBody.tsx:46]: why do you need both "iconFlag" and "bottomIcon"?

File: src/components/Table/TableBody.tsx [hunk 2]
```diff
@@ -1,69 +1,85 @@
...
+            description={pokemonDetails.description}
+            profileAttributes={transformPokemonDataToProfileAttributes(
+              pokemonDetails
+            )}
+            statsAttributes={transformPokemonDataToStatsAttributes(
+              pokemonDetails
+            )}
           />
         )}
+        {isLoading && <p>Loading...</p>}
```
Comment (YuvalLeviMoveoHLS) [inline src/components/Table/TableBody.tsx]: use skeleton

File: src/features/cardView/PokemonBasicCard/PokemonBasicCard.tsx [hunk 3]
```diff
@@ -1,33 +1,31 @@
 import { StyledBasicCard } from "./styles";
 import { CSSProperties } from "styled-components";
 import CardBody from "../../../components/CardBody/CardBody";
+import { IPokemonData } from "../../../types/pokemonTypes";
+import pokeballIcon from "../../../assets/icons/pokeball.svg";
 
 interface BasicCardProps {
```
Comment (YuvalLeviMoveoHLS) [inline src/features/cardView/PokemonBasicCard/PokemonBasicCard.tsx]: PokemonBasicCardProps

File: src/hooks/useFetchPokemonData.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
+import axios from "axios";
+import { SortByValues } from "../constants/tableSortbyOptions";
+import { IPokemonData } from "../types/pokemonTypes";
+
+const URL = "http://localhost:3000/api/pokemons";
```
Comment (YuvalLeviMoveoHLS) [inline src/hooks/useFetchPokemonData.ts]: move to env file

File: src/hooks/useFetchPokemonData.ts [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  isOwned: boolean;
+  searchValue: string;
+  sortBy: SortByValues | "";
+  page: number;
+  limit: number;
+}): Promise<{ pokemons: IPokemonData[]; totalCount: number }> => {
+  try {
+    const endpoint = isOwned
+      ? "/search-pokemons?isOwned=true"
+      : "/search-pokemons";
```
Comment (YuvalLeviMoveoHLS) [inline src/hooks/useFetchPokemonData.ts]: add isOwned to the axios request params

File: src/features/cardView/PokemonCardView/SkeletonCardView.tsx [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+interface SkeletonCardViewProps {
+  itemsPerPage?: number;
+}
+
+const SkeletonCardView = ({ itemsPerPage = 8 }: SkeletonCardViewProps) => {
+  const skeletonCards = Array.from(new Array(itemsPerPage));
+
+  return (
+    <Box display="flex" flexWrap="wrap" gap={"24px"}>
+      {skeletonCards.map((_, index) => (
```
Comment (YuvalLeviMoveoHLS) [inline src/features/cardView/PokemonCardView/SkeletonCardView.tsx]: use cssSpacings

File: src/features/cardView/PokemonCardView/SkeletonCardView.tsx [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+        <Box key={index} width={322} height={314}>
+          <Skeleton
+            variant="rectangular"
+            width={322}
+            height={314}
+            sx={{ borderRadius: "8px" }}
+          />
```
Comment (YuvalLeviMoveoHLS) [inline src/features/cardView/PokemonCardView/SkeletonCardView.tsx]: use  cssSpacings

File: src/features/cardView/PokemonCardView/SkeletonCardView.tsx [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+  const skeletonCards = Array.from(new Array(itemsPerPage));
+
+  return (
+    <Box display="flex" flexWrap="wrap" gap={"24px"}>
+      {skeletonCards.map((_, index) => (
+        <Box key={index} width={322} height={314}>
+          <Skeleton
+            variant="rectangular"
+            width={322}
+            height={314}
```
Comment (YuvalLeviMoveoHLS) [inline src/features/cardView/PokemonCardView/SkeletonCardView.tsx]: add width and height to SkeletonCardViewProps

File: src/hooks/useHandleFight.ts [hunk 9]
```diff
@@ -0,0 +1,99 @@
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import axios from "axios";
+import { IPokemonData } from "../types/pokemonTypes";
+
+const URL = "http://localhost:3000/api/battle";
+
```
Comment (YuvalLeviMoveoHLS) [inline src/hooks/useHandleFight.ts]: move to env file

**Review by YuvalLeviMoveoHLS — CHANGES_REQUESTED** [review]

File: src/components/CardBody/CardBody.tsx (hunk 1 above)
Comment (IlanSirisky) [reply src/components/CardBody/CardBody.tsx:46]: iconFlag for if pokemon is owned, bottomIcon to check if prop was actually given.

File: src/features/cardView/PokemonCardView/SkeletonCardView.tsx [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+  const skeletonCards = Array.from(new Array(itemsPerPage));
+
+  return (
+    <Box display="flex" flexWrap="wrap" gap={cssSpacings.s24}>
+      {skeletonCards.map((_, index) => (
+        <Box key={index} width={customWidth} height={customHeight}>
+          <Skeleton
+            variant="rectangular"
+            width={322}
+            height={314}
```
Comment (YuvalLeviMoveoHLS) [inline src/features/cardView/PokemonCardView/SkeletonCardView.tsx]: width={customWidth}
height={customWidth}

**Review by YuvalLeviMoveoHLS — CHANGES_REQUESTED** [review]

**Pull request merged**